### PR TITLE
Add `plone.autoform` documentation for classic UI

### PR DESCRIPTION
**Pull request by petschki**
Put `plone.autoform` documentation to the designated places.

### CODE REVIEW

File: docs/classic-ui/forms.md
Comment (MrTango) [inline docs/classic-ui/forms.md]: this is already covert in the widgets chapter and fits better there.
In general, we should try to place individual parts of autoform docs into there fitting chapters.
Could be anything from schemas, fields, widgets to forms and content types.
As content types and forms are both sharing schemas, fields and widgets, I would first try to fit in one of those and reference it from forms and content types where it make sense.

File: docs/classic-ui/forms.md
Comment (MrTango) [inline docs/classic-ui/forms.md]: chapter: fields feels more like the right place for this with a reference from here maybe.

File: docs/classic-ui/forms.md
Comment (MrTango) [inline docs/classic-ui/forms.md]: chapter: schema feels more like the right place, see explanations below.

File: docs/classic-ui/forms.md
Comment (MrTango) [inline docs/classic-ui/forms.md]: chapter: schema feels more like the right place, see explanations below.

File: docs/classic-ui/forms.md
Comment (MrTango) [inline docs/classic-ui/forms.md]: chapter: schema feels more like the right place, see explanations below.

File: docs/classic-ui/forms.md
Comment (MrTango) [inline docs/classic-ui/forms.md]: chapter: widgets feels more like the right place, see explanations below.

**Review by MrTango — CHANGES_REQUESTED** [review]

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
A field's widget can be displayed in several modes:
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Style guide suggestions.

- Use `-` for list items.
- Indent 4 spaces.

This is more of a definition list, than a list.

```suggestion
`hidden`
:   A record of the field's value that is included only in the HTML source
```

Would you please apply the definition list pattern to these list items?

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Style guide: one line per sentence.
```suggestion
In this case the mode for the `secret` field is set to `hidden` for most forms, but `input` for forms that provide the `IEditForm` interface.
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Dedent the example to avoid horizontal scrolling.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Dedent the example to avoid horizontal scrolling.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: One sentence per line.

Inline literal.

```suggestion
In other words, `form:mode` may be either a single mode, or a space-separated list of `form_interface:mode` pairs.
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md:125]: Don't use -ing suffix for active verbs in headings.

```suggestion

## Control form presentation
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
### Change a field's display mode
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
### Omit fields
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: One sentence per line, and make IEditForm an inline literal:

```
`IEditForm`
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Replace `'` with `"`, per Black.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Dedent example

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
### Reorder fields
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
`form:omitted` may be either a single boolean value, or a space-separated list of `form_interface:boolean` pairs.
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: One sentence per line.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Replace `'` with `"`, per Black.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: One sentence per line.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
The value passed to the directive may be either `*` (indicating before or after
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
field in the current schema or a base schema. Prefix with the schema name (for example,
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
`IDublinCore.title`) to refer to a field in another schema. Use an
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Replace `'` with `"`, per Black.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
In supermodel XML, fieldsets are specified by grouping fields within a
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: One sentence per line.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md:138]: Dedent

File: docs/classic-ui/forms.md
Comment (stevepiercy) [reply docs/classic-ui/forms.md]: I agree with @MrTango. This section already exists in https://6.docs.plone.org/backend/widgets.html#changing-a-field-s-widget. I won't review its content, as the existing content has already been reviewed and deployed.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: All the above, one sentence per line.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
Zope 3-style name (such as `cmf.ManagePortal` instead of `Manage portal`).
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: Dedent both above code examples

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]: One sentence per line.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md:155]: Dedent.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md:169]: One sentence per line, all the above.

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
-   `view/w` is a dictionary of all widgets, including those from non-default
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
-   `view/fieldsets` is a dictionary of all fieldsets (not including the
```

File: docs/classic-ui/forms.md
Comment (stevepiercy) [inline docs/classic-ui/forms.md]:
```suggestion
  default fieldset, in other words, those widgets not placed into a fieldset. The keys
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Mostly a bunch of style guide suggestions.

- Please use one line per sentence. Documentation is not code.
- Dedent code examples to avoid horizontal scrolling.
- Some missing inline literals.

Also I agree with and defer to @MrTango suggestions for where topics belong. I would replace the content with reference links to the authoritative content. That might require adding references and  targets in MyST syntax.

For example:

````
```{seealso}
{ref}`Referenced section <target>`
```
````

and in the target section:

````
```
(target)=
```
````

**Comment by petschki** [comment]
I've also removed this part:

https://github.com/plone/documentation/pull/1571/files#diff-2a092039aef3cb61f8a3fdc888afaaca68f5dd8a64b6b612cbf3c3d2af1b01d4L591-L603

`plone.directives.form` is outdated and the default value hint is already covered here https://github.com/plone/documentation/pull/1571/files#diff-2a092039aef3cb61f8a3fdc888afaaca68f5dd8a64b6b612cbf3c3d2af1b01d4L591-L603

File: docs/backend/schemas.md
Comment (stevepiercy) [inline docs/backend/schemas.md]: Do we still need the `u` prefix for Plone 6? I think we can remove these instances, correct?

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Do we need the `u` in `u"value"` in Plone 6? I would prefer to remove all these instances, if not needed.

Otherwise I pushed a few commits to tidy this up and push it over the finish line. It requires a core contributor or maintainer to review its technical accuracy.

File: docs/backend/schemas.md
Comment (petschki) [reply docs/backend/schemas.md]: nope ... since python3 no more `u""` is needed

Comment (stevepiercy) [reply docs/backend/schemas.md]: Thanks! Fixed in [5dbcb72](https://github.com/plone/documentation/pull/1571/commits/5dbcb72a115ce703945d36093c79c2e0b5afeaa2)

File: docs/backend/schemas.md
Comment (jensens) [inline docs/backend/schemas.md]: is the dot by intend?

**Review by jensens — DISMISSED** [review]
Very important! LGTM, just one question.

File: docs/backend/schemas.md
Comment (stevepiercy) [reply docs/backend/schemas.md]: If it is intentional, then it should be formatted as:

`.<fieldname>` or `.<field_name>`

...and perhaps using snake_case is more legible?

Comment (petschki) [reply docs/backend/schemas.md]: I've fixed that

**Review by stevepiercy — APPROVED** [review]
LGTM. One more pass from @jensens, or anyone else, and we can merge. Thank you!

**Comment by MrTango** [comment]
> I've also removed this part:
> 
> https://github.com/plone/documentation/pull/1571/files#diff-2a092039aef3cb61f8a3fdc888afaaca68f5dd8a64b6b612cbf3c3d2af1b01d4L591-L603
> 
> `plone.directives.form` is outdated and the default value hint is already covered here https://github.com/plone/documentation/pull/1571/files#diff-2a092039aef3cb61f8a3fdc888afaaca68f5dd8a64b6b612cbf3c3d2af1b01d4L591-L603

i can't find the default value info, you want to remove, elsewhere.
Where do you see them?

**Review by MrTango — APPROVED** [review]
Except the default value part you want to remove, LGTM. feel free to merge

**Comment by stevepiercy** [comment]
> Except the default value part you want to remove, LGTM. feel free to merge

@MrTango do you mean that you want @petschki to restore the default value part that he removed? I'm confused because you also said "feel free to merge". Please clarify.

**Comment by MrTango** [comment]
> > Except the default value part you want to remove, LGTM. feel free to merge
> 
> @MrTango do you mean that you want @petschki to restore the default value part that he removed? I'm confused because you also said "feel free to merge". Please clarify.

yes, he said that what he want's to remove is described elsewhere, but i could not find it. 
So either enlighten me and remove it or keep it.
But let him do it, no need to act for you here.

**Comment by petschki** [comment]
Sorry for the confusion. I've only de-duplicated the "default value hint" in `schemas.md` and remove the old link to `plone.directives.form` ... the hint is already in the warning box here https://6.docs.plone.org/backend/schemas.html?highlight=default%20value#field-constructor-parameters ... so all good from my side to merge.

**Comment by stevepiercy** [comment]
@petschki thanks for the clarification. I'll merge when it is green. Thank you!

**Comment by stevepiercy** [comment]
There was a broken link to the TinyMCE 4 to 5 upgrade guide. I also created a new issue to upgrade to the latest version TinyMCE 7, as v5 is EOL. https://github.com/plone/Products.CMFPlone/issues/3936

**Pull request merged**
